smarthint/loaders/productListingPage.ts

```diff
@@ -0,0 +1,177 @@
+import { ProductListingPage } from "../../commerce/types.ts";
+import { AppContext } from "../mod.ts";
+import {
+  getFilterParam,
+  getPaginationInfo,
+  getSortParam,
+  resolvePage,
+  toFilters,
+  toProduct,
+  toSortOption,
+} from "../utils/transform.ts";
+import { redirect } from "deco/mod.ts";
+import { getSessionCookie } from "../utils/getSession.ts";
+import { FilterProp, SearchSort, SHProduct } from "../utils/typings.ts";
+import { RuleType } from "./PLPBanners.ts";
+import { getCategoriesParam } from "./recommendations.ts";
+
+export interface Props {
+  /**
+   * @hide
+   */
+  term?: string;
+  /**
+   * @description Number of products that must be returned per page
+   */
+  size: number;
+  /**
+   * @hide
+   */
+  searchSort?: SearchSort;
+  /**
+   * @hide
+   */
+  filter?: FilterProp[];
+  /**
+   * @hide
+   */
+  from?: number;
+  /**
+   * @hide
+   */
+  rule?: string;
+  /**
+   * @hide
+   */
+  ruletype?: RuleType;
+  /**
+   * @hide
+   */
+  condition?: {
+    field?: string;
+    value?: string;
+    validation?: string;
+  };
```

Comment:
can you add a JS comment explaining why these properties are hidden?

Reply:
SH colaborators ask me to remove this props from admin and this props should be generated by the loaders itself. By receiving these props, we can use the loader inline for infinite scroll or other functionalities.

Reply:
I think what Candeia said is to add the comment at the code.